spec/operators/withLatestFrom-spec.ts

```diff
@@ -1,279 +1,318 @@
+        x: 'bfj',
+        y: 'cgk',
+        z: 'dhl',
+      };
+      const project = (a: string, b: string, c: string) => a + b + c;
+
+      const result = e1.pipe(
+        mergeMap((x: string) => of(x)),
+        withLatestFrom(e2, e3, project),
+        mergeMap((x: string) => of(x))
+      );
+
+      expectObservable(result, unsub).toBe(expected, values);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+      expectSubscriptions(e3.subscriptions).toBe(e3subs);
+    });
   });
 
   it('should handle empty', () => {
-    const e1 =   cold(    '|');
-    const e1subs =        '(^!)';
-    const e2 =   hot('--e--^-f---g---h----|');
-    const e2subs =        '(^!)';
-    const e3 =   hot('--i--^-j---k---l----|');
-    const e3subs =        '(^!)';
-    const expected =      '|'; // empty
-
-    const result = e1.pipe(withLatestFrom(e2, e3));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
-    expectSubscriptions(e3.subscriptions).toBe(e3subs);
+    rxTestScheduler.run(({ hot, cold, expectObservable, expectSubscriptions }) => {
+      const e2 = hot('  --e--^-f---g---h----|');
+      const e2subs = '       (^!)            ';
+      const e3 = hot('  --i--^-j---k---l----|');
+      const e3subs = '       (^!)            ';
+      const e1 = cold('      |               ');
+      const e1subs = '       (^!)            ';
+      const expected = '     |               '; // empty
+
+      const result = e1.pipe(withLatestFrom(e2, e3));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+      expectSubscriptions(e3.subscriptions).toBe(e3subs);
+    });
   });
 
   it('should handle never', () => {
-    const e1 =   cold('-');
-    const e1subs =        '^               ';
-    const e2 =   hot('--e--^-f---g---h----|');
-    const e2subs =        '^              !';
-    const e3 =   hot('--i--^-j---k---l----|');
-    const e3subs =        '^              !';
-    const expected =      '--------------------'; // never
-
-    const result = e1.pipe(withLatestFrom(e2, e3));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
-    expectSubscriptions(e3.subscriptions).toBe(e3subs);
+    rxTestScheduler.run(({ cold, hot, expectObservable, expectSubscriptions }) => {
+      const e2 = hot('   --e--^-f---g---h----|  ');
+      const e2subs = '       ^--------------!   ';
+      const e3 = hot('  --i--^-j---k---l----|   ');
+      const e3subs = '       ^--------------!   ';
+      const e1 = cold('        -                ');
+      const e1subs = '         ^----------------';
+      const expected = '    --------------------'; // never
```

Comment:
nitpick: `^` characters that represent the zero frames of the hot observables - the subscriptions to them - are not aligned.

Reply:
Good catch. It is fixed now.